### PR TITLE
feat(notifications): Endpoint to get all resources for configured notification

### PR DESCRIPTION
**Pull request by dmyios**
## Description

GET `/configuration/notifications/resources`

**Fixes** # MON-20151

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -17921,3 +17921,9 @@ msgstr "Le nom d'hôte ne doit pas commencer par '_Module_'"
 
 msgid "Error while retrieving a host"
 msgstr "Error lors de la récupération d'un hôte"
+
+msgid "You are not allowed to list notification resources"
```
Comment (abuathier) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17925]: Missing placeholders in other messages.po files

**Review by abuathier — APPROVED** [review]

**Review by abuathier — APPROVED** [review]

File: centreon/doc/API/v23.10/Configuration/Notification/FindNotifiableResources.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+get:
+  tags:
+    - Notification
+  summary: "List all resources and events triggering a notification rule"
+  description: |
+    Return all resources: hosts, services, business activities and configured events
+    trigerring a notification rule.
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/FindNotifiableResources.yaml:7]:
```suggestion
    triggering a notification rule.
```

File: centreon/tests/php/Core/Notification/Application/UseCase/FindNotifiableResources/FindNotifiableResourcesTest.php [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+    $useCase($this->presenter, $requestUid);
+
+    expect($this->presenter->responseStatus)
+        ->toBeInstanceOf(ForbiddenResponse::class)
+        ->and($this->presenter->responseStatus->getMessage())
+        ->toBe(NotificationException::listResourcesNotAllowed()->getMessage());
+});
+
+it('should present a Not Found Response when there\'re no notifiable resources.', function () {
```
Comment (cg-tw) [inline centreon/tests/php/Core/Notification/Application/UseCase/FindNotifiableResources/FindNotifiableResourcesTest.php:60]:
```suggestion
it('should present a Not Found Response when there are no notifiable resources.', function () {
```

**Review by abuathier — DISMISSED** [review]

**Review by cg-tw — DISMISSED** [review]

File: centreon/doc/API/centreon-api-v23.10.yaml [hunk 4]
```diff
@@ -256,6 +256,8 @@ paths:
     $ref: "./v23.10/Configuration/Notification/DeleteNotifications.yaml"
   /configuration/notifications/{notification_id}:
     $ref: "./v23.10/Configuration/Notification/Notification.yaml"
+  /configuration/notifications/resources:
+    $ref: "./v23.10/Configuration/Notification/FindNotifiableResources.yaml
```
Comment (dmyios) [inline centreon/doc/API/centreon-api-v23.10.yaml:260]:
```suggestion
    $ref: "./v23.10/Configuration/Notification/FindNotifiableResources.yaml"
```

**Review by cg-tw — APPROVED** [review]

**Review by abuathier — DISMISSED** [review]

**Review by cg-tw — DISMISSED** [review]

File: centreon/src/Core/Notification/Infrastructure/Repository/DbNotifiableResourceFactory.php [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+                continue;
+            }
+
+            if ($currentHostId === $record['host_id']) {
+                $currentRecords[] = $record;
+                $index++;
+                continue;
+            }
+
+            if ($record['host_events'] !== '0') {
```
Comment (adr-mo) [inline centreon/src/Core/Notification/Infrastructure/Repository/DbNotifiableResourceFactory.php:118]: Please could you replace the '0' by a constant to make code clearer

Comment (dmyios) [reply centreon/src/Core/Notification/Infrastructure/Repository/DbNotifiableResourceFactory.php]: Done

**Review by abuathier — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**